src/pages/admin/donate/components/input/Input.tsx

```diff
@@ -0,0 +1,137 @@
+    name: string;
+    value?: string;
+    editable?: boolean;
+    handleChange?: (e: React.ChangeEvent<HTMLTextAreaElement>) => void;
+    handleBlur?: (e: React.FocusEvent<HTMLTextAreaElement>) => void;
+    onValueChange?: (val: string) => void;
+    onlyNumbers?: boolean;
+    maxLength?: number;
+    className?: string;
+}
+
+export const Input = ({
+    label,
+    placeholder = DONATE_TEXT.PLACEHOLDER.DEFAULT,
+    isTitle = false,
+    isRequired,
+    prefix = '',
+    name,
+    value: externalValue,
+    editable = true,
+    handleChange,
+    handleBlur,
+    onValueChange,
+    onlyNumbers = false,
+    maxLength,
+    className,
+}: InputProps) => {
+    const computedInitialValue =
+        externalValue !== undefined && externalValue !== null ? prefix + externalValue.replace(prefix, '') : prefix;
+
+    const [value, setValue] = useState(computedInitialValue);
+    const [initialValue, setInitialValue] = useState(computedInitialValue);
+    const [hasEdited, setHasEdited] = useState(false);
+    const [isFocused, setIsFocused] = useState(false);
+    const textAreaRef = useRef<HTMLTextAreaElement>(null);
+
+    useEffect(() => {
+        const newValue =
+            externalValue !== undefined && externalValue !== null ? prefix + externalValue.replace(prefix, '') : prefix;
+
+        setValue(newValue);
+        setInitialValue(newValue);
+    }, [externalValue, prefix]);
+
+    useEffect(() => {
+        if (!textAreaRef.current) return;
+
+        const ta = textAreaRef.current;
+        const style = window.getComputedStyle(ta);
+        const minHeight = parseFloat(style.minHeight);
+
+        if (value && value !== prefix && ta.scrollHeight > minHeight) {
+            ta.style.height = style.minHeight;
+            ta.style.height = `${ta.scrollHeight}px`;
+        }
+    }, [value, prefix]);
+
+    const onChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+        let newValue = e.target.value;
+
+        if (onlyNumbers) {
+            const numbers = newValue.replace(/\D/g, '');
+            newValue = prefix ? prefix + numbers : numbers;
+        }
+        setValue(newValue);
+
+        onValueChange?.(newValue);
+
+        if (!hasEdited && newValue !== initialValue) {
+            setHasEdited(true);
+        }
+
+        handleChange?.(e);
+    };
+
+    const handleClear = () => {
+        setValue(prefix);
+        onValueChange?.(prefix);
+        setHasEdited(true);
+    };
+
+    const showClearButton = isFocused && value.length > prefix.length;
+
+    const hasValue = value && value !== prefix;
+
+    return (
+        <div className={`input ${isTitle ? 'input-title' : ''} ${hasEdited ? 'input-changed' : ''} ${className ?? ''}`}>
+            {label && (
+                <div className={isTitle ? 'input-title-label' : 'input-label'}>
+                    {isRequired && editable && <span className="input-required">*</span>}
+                    {!isTitle && label}
+                </div>
```

Comment:
Should be label here?
```suggestion
                <label className={isTitle ? 'input-title-label' : 'input-label'}>
                    {isRequired && editable && <span className="input-required">*</span>}
                    {!isTitle && label}
                </label>
```